Scripts/AudioPlayer.cs

```diff
@@ -0,0 +1,66 @@
+using Godot;
+using System;
+
+public partial class AudioPlayer : AudioStreamPlayer
+{
+	// Konstanten müssen einen Datentyp haben, daher wird der Typ `AudioStream` verwendet
+	private static readonly AudioStreamMP3 titleMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/MainMenu.mp3");
+	private static readonly AudioStreamMP3 mainMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/MainSzene.mp3");
+	private static readonly AudioStreamMP3 greenMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/Greenhouse.mp3");
+	private static readonly AudioStreamMP3 minigameMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/Minigames.mp3");
+	private static readonly AudioStreamMP3 memoryMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/mini games/Memory.mp3");
+	private static readonly AudioStreamMP3 shopMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/Shop.mp3");
+	private static readonly AudioStreamMP3 flappyMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/mini games/FlappyPlant.mp3");
+	private static readonly AudioStreamMP3 wormMusic = (AudioStreamMP3) GD.Load("res://Sounds/Background Music/mini games/Worm.mp3");
+
+	// Methode zum Abspielen von Musik mit einem optionalen Lautstärkewert
+	private void _play_music(AudioStream music, float volume = 0.0f)
+	{
+		if (Stream == music)
+		{
+			return;
+		}
+		Stream = music;
+		VolumeDb = volume;
+		if (music is AudioStreamMP3 mp3Stream)
+		{
+			mp3Stream.Loop = true;
+		}
+		Play();
+	}
```

Comment:
In all references to this function it is always used with a AudioStreamMP3 as the music parameter. Therefore the check in line 25 becomes obsolete.

I propose the following refactoring of this function:

```suggestion
	private void _play_music(AudioStreamMP3 music, float volume = 0.0f)
	{
		if (Stream == music)
		{
			return;
		}
		Stream = music;
		VolumeDb = volume;

		music.Loop = true;
		
		Play();
	}
```